app/auth/ValidatedActions.scala

```diff
@@ -0,0 +1,32 @@
+package auth
```

Comment:
Sorry, I should have spotted this before: I think this validation is conceptually different to the authorisation, even if the steps are similar (i.e. check something exists in the API, throw an error if not).

This is more about the current state of the transaction. I don't quite know how we should structure the code to handle that at the moment - whether it should wrap secureAction, or if it should be at a different level of abstraction. I think that will get clearer as we build up the steps of the upload process. But it would be good to at least put it into a different module to auth, so that we don't confuse state errors with auth errors.

Reply:
Yep, that makes sense. As you say, it's pretty tightly coupled into the secureAction at the moment so any major changes will take time but I've moved it into a new validation package.